dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java

```diff
@@ -37,6 +58,54 @@ public JooqFactory getJooqFactory(C configuration) {
     }
 
     public Configuration getConfiguration() {
-        return configuration;
+        return jooqFactoryConfigurationMap.values().stream().findFirst().orElse(null);
+    }
+
+    public Map<String,Configuration> getConfigurationMap() {
+        return jooqFactoryConfigurationMap;
+    }
+
+    /**
+     * Override to configure the name of the bundle, if there is only one data source.
+     * Used for the bundle health check and database pool metrics.
+     *
+     * @return {#DEFAULT_NAME} by default.
+     */
+    protected String name() {
+        return DEFAULT_NAME;
+    }
+
+    /**
+     * Override to set a default DataSourceFactory.
+     *
+     * @return the default DataSourceFactory
+     */
+    protected DataSourceFactory getDefaultDataSourceFactory(C configuration) {
+        return getDataSourceFactory(configuration);
+    }
+
+    /**
+     * Override to add multiple data source factories to the bundle, each distinguished by a unique name.
+     *
+     * @param configuration a Configuration
+     * @return a Map containing DataSourceFactories, with the key being a String name.
+     * This name is referenced from the @DataSourceParam(name = "stringKey") annotation.
+     */
+    protected Map<String,DataSourceFactory> getDataSourceFactories(C configuration) {
+        final Map<String,DataSourceFactory> dataSourceFactoryMap = new HashMap<>();
+        dataSourceFactoryMap.put(name(), getDataSourceFactory(configuration));
+        return dataSourceFactoryMap;
+    }
+
+    private Configuration getDefaultJooqFactoryConfiguration(
+            DataSourceFactory defaultDataSourceFactory,
+            Map<String,DataSourceFactory> dataSourceFactoryMap) {
+        Configuration result = null;
```

Comment:
I experimented with enums and was almost able to get it to work nicely, except for a couple of issues: 
- the @Named annotation must take a constant value, and an Enum.toString() method is not constant, so I can't do something like this: @Named(DataSources.MASTER.toString())
- passing in a parameterized type of <E extends Enum> into JooqBundle will work for the class instantiating JooqBundle, but not for JooqBundle calling it's own default overriden method: DataSourceEnum<E, C> getDataSourceFactories(C configuration):
```
    @Override
    public DataSourceEnum<E, C> getDataSourceFactories(C configuration) {
        return new DataSourceEnum<C>() {
            @Override
            public LinkedHashMap<E, DataSourceFactory> getDataSourceFactoryMap(C configuration) {
                final LinkedHashMap<E, DataSourceFactory> dataSourceFactoryMap = new LinkedHashMap<>();
                dataSourceFactoryMap.put(/* what  to put in here? */, getDataSourceFactory(configuration));
                return dataSourceFactoryMap;
            }
        };
    }
```